### DISCUSSION TITLE
Why doesn't this button behave like the span?

### DISCUSSION THREAD

**Discussion by frederikhors** [tailwindlabs/tailwindcss#10246] [Help] [answered]
In the below code (REPL here: https://play.tailwindcss.com/0sbo1393b3) the only thing that changes between these two elements is:

```
span
```

instead of

```
button
```

What I need is the button fill all the navbar height, like the span one (no red background visible).

Why it does not?

```html
<div class="top-0 flex h-20 w-full justify-between bg-white">
  <span class="flex bg-red-400">
    <a class="flex items-center bg-green-400 px-6">
      <span>Hello</span>
    </a>
  </span>

  <button class="flex bg-red-400">
    <a class="flex items-center bg-green-400 px-6">
      <span>Hello</span>
    </a>
  </button>
</div>
```

**Comment by wongjn** [comment, marked as answer]
At least in Chrome, [the user agent stylesheet applies `align-items: start` to `<button>` elements](https://chromium.googlesource.com/chromium/blink/+/8d2d7506bebf376fcb885f8344e1df38de2998f2/Source/core/css/html.css#587), where as the default value is `stretch` (as can be seen on the `<span>` element).

To achieve identical behavior, consider apply `align-items: start` to the `<button>` element via the `items-stretch` class, or apply `align-self: stretch` via the `self-stretch` class to the child `<a>` element.

**Reply by frederikhors** [reply]
Thanks.

`flex items-stretch` on button works. `self-stretch` on `<a>` doesn't.

I found another way too: using `h-full` on `<a>`, what do you think?

**Reply by wongjn** [reply]
I think you are incorrect – `self-stretch` on `<a>` should work: https://play.tailwindcss.com/JVHNUHIk4g

**Reply by frederikhors** [reply]
It works. You're right. What do you think about `h-full` on `<a>`?

**Reply by wongjn** [reply]
I don't have any particular opinion on it.
